src/Haskell/Ide/Engine/Plugin/Generic.hs

```diff
@@ -126,9 +93,13 @@ pureTypeCmd newPos tm info =
 
     f (range', t) =
       case oldRangeToNew info range' of
-        (Just range) -> [(range , T.pack $ GM.pretty dflag st t)]
+        (Just range) -> [(range , T.pack $ prettyTy st t)]
         _ -> []
 
+    prettyTy stl
+      = showOneLine dflag stl . pprTypeForUser
+
+-- TODO: MP: Why is this defined here?
```

Comment:
Did someone figure this out?

Reply:
I think this is defined here because the Ord Instance of `Range` is auto-derived and not what we want here. We should upstream this function, imo, and define a more suited Ord instance for Range.

Reply:
Not a more suited Ord Instance, but still upstream this function.